### DISCUSSION TITLE
reverse dependencies show dependent only once even if depending multiple times

### DISCUSSION THREAD

**Discussion by rursprung** [rust-lang/crates.io#7513] [Feature Requests]
### Current Behavior

if a crate has several dependencies to another crate (with aliases) then it is still only shown once in the reverse dependencies list instead of once per dependency.

this constellation usually happens when one crate depends on multiple major releases of another crate - the use-case which i know is mostly in the embedded world, where a lot of drivers, HALs, etc. implement the [`embedded-hal`](https://crates.io/crates/embedded-hal) APIs for both 0.2.x (current "stable" release) and 1.0.0-alpha.x or 1.0.0-rc.x (an example would be [`stm32f4xx-hal`](https://crates.io/crates/stm32f4xx-hal)).
in the dependencies list of this crate the dependency is correctly shown twice, however in the reverse dependencies of embedded-hal it is only shown once.

### Expected Behavior

the dependent crate is listed once per dependency, with the correct information (which version it depends on)

### Steps To Reproduce

1. go to [the reverse dependencies of `embedded-hal`](https://crates.io/crates/embedded-hal/reverse_dependencies)
2. try to see that `stm32f4xx-hal` is listed twice: once for v1.0.0-alpha.x and once for v0.2.x

### Environment

n/a

### Anything else?

this is related to my new feature request: https://github.com/rust-lang/crates.io/discussions/7511

**Comment by Turbo87** [comment]
the reverse dependencies endpoint is our most costly endpoint in terms of performance already. if we would make the database query even more complex I don't think we could keep offering this endpoint at all. 😐
